phoenix-core/src/test/java/org/apache/phoenix/query/PhoenixTestBuilder.java

```diff
@@ -212,66 +254,142 @@ private static String getFQColumnsAsString(List<String> columns, List<String> fa
-        public static List<String> TENANT_VIEW_COLUMN_FAMILIES = asList(null, null, null);
+        // Read batch of rows
+        public void readRows() throws SQLException {
+            dataTable.clear();
+            String sql = getDML();
+            Connection connection = getConnection();
+            try (Statement stmt = connection.createStatement()) {
+
+                final PhoenixStatement pstmt = stmt.unwrap(PhoenixStatement.class);
+                ResultSet rs = pstmt.executeQuery(sql);
+                List<String> cols = getValidationColumns();
+                List<Object> values = Lists.newArrayList();
+                Set<String> rowKeys = getRowKeyColumns() == null || getRowKeyColumns().isEmpty() ?
+                        Sets.<String>newHashSet() :
+                        Sets.newHashSet(getRowKeyColumns());
+                List<String> rowKeyParts = Lists.newArrayList();
+                while (rs.next()) {
+                    for (String col : cols) {
+                        Object val = rs.getObject(col);
+                        values.add(val);
+                        if (rowKeys.isEmpty()) {
+                            rowKeyParts.add(val.toString());
+                        }
+                        else if (rowKeys.contains(col)) {
+                            rowKeyParts.add(val.toString());
+                        }
+                    }
 
-        public static List<String> TABLE_PK_COLUMNS = asList("OID", "KP");
-        public static List<String> GLOBAL_VIEW_PK_COLUMNS = asList("ID");
-        public static List<String> TENANT_VIEW_PK_COLUMNS = asList("ZID");
+                    String rowKey = Joiner.on("-").join(rowKeyParts);
+                    for (int v = 0; v < values.size(); v++) {
+                        dataTable.put(rowKey,cols.get(v), values.get(v));
+                    }
+                    values.clear();
+                    rowKeyParts.clear();
+                }
+                LOGGER.info(String.format("########## rows: %d", dataTable.rowKeySet().size()));
 
-        public static List<String> TABLE_INDEX_COLUMNS = asList("COL1");
-        public static List<String> TABLE_INCLUDE_COLUMNS = asList("COL3");
+            } catch (SQLException e) {
+                LOGGER.error(String.format(" Error [%s] initializing Reader. ",
+                        e.getMessage()));
+                throw e;
+            }
+        }
+    }
 
-        public static List<String> GLOBAL_VIEW_INDEX_COLUMNS = asList("COL4");
-        public static List<String> GLOBAL_VIEW_INCLUDE_COLUMNS = asList("COL6");
+    // An implementation of the DataReader.
+    public static class BasicDataReader extends AbstractDataReader {
 
-        public static List<String> TENANT_VIEW_INDEX_COLUMNS = asList("COL9");
-        public static List<String> TENANT_VIEW_INCLUDE_COLUMNS = asList("COL7");
+        Connection connection;
```

Comment:
I might be missing something, but it looks like we create a local variable `Connection connection = ... ` everywhere. Is this class variable required?

Reply:
The test cases create the connection and call the setter method (setConnection)

Reply:
Got it. I misread as a call to DriverManager.getConnection() instead of getConnection.